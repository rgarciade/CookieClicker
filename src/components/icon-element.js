import { LitElement, html } from 'lit';
import 'material-design-icons-iconfont/dist/material-design-icons.css';
import { iconStyle } from './styles/icon-style.js';
export class IconElement extends LitElement {
	/**
	 * @property {String} icon - Icon name from material design.
	 */
	static get properties() {
		return {
			icon: { type: String },
		};
	}
	static get styles() {
		return [iconStyle];
	}

	constructor() {
		super();
	}
	render() {
		return html` <i class="material-icons">${this.icon}</i> `;
	}
}
window.customElements.define('icon-element', IconElement);
