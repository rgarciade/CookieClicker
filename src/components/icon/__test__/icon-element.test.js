import { html, fixture } from '@open-wc/testing';
import '../icon-element.js';

import { expect } from '@jest/globals';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanedLit } from '../../../../testFunctions/cleanedLit.js';
expect.extend(toHaveNoViolations);

describe('icon-element', () => {
	it('icon matches the snapshot', async () => {
		const el = await fixture(html`<icon-element icon="person"></icon-element>`);
		const dataCleaned = cleanedLit(el);
		expect(dataCleaned).toMatchSnapshot();
	});
	it('accessibility test', async () => {
		const el = await fixture(
			html`<main><icon-element icon="person"></icon-element></main>`
		);
		const results = await axe(el);
		expect(results).toHaveNoViolations();
	});
});
