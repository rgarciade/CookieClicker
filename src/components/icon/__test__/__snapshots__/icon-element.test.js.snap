// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon-element icon matches the snapshot 1`] = `
"[36m<icon-element[39m
  [33micon[39m=[32m"person"[39m
[36m/>[39m"
`;
